docs/_static/js/search.js

```diff
@@ -0,0 +1,269 @@
+/**
+ * Get version of the currently served docs.
+ *
+ * PR builds have their version set to the PR ID (for example "6753").
+ * If the docs are built for a PR, use the "latest" search index.
+ * Otherwise, use the search index for the current version.
+ */
+function getReadTheDocsVersion() {
+  const RTD_VERSION = (window.READTHEDOCS_DATA || {}).version || 'latest';
+  const version = RTD_VERSION.match(/^\d+$/) ? 'latest' : RTD_VERSION;
+  return version;
+}
+
+function getVersionFacetFilter() {
+  return `version:${getReadTheDocsVersion()}`;
+}
+
+/**
+ * Return true (debug: on) for local builds or Read the Docs PR previews.
+ *
+ * The debug mode allows inspection of the dropodown.
+ */
+function getSearchDebugMode() {
+  let debug = false;
+  if (window.READTHEDOCS_DATA === undefined) {
+    // When developing locally, the `window.READTHEDOCS_DATA` object does not exist.
+    debug = true;
+  } else {
+    // When PR preview on Readthedocs, then the version can be converted into
+    // a number. This does not work for the production version identifiers
+    // like 'stable', 'latest', 'v2.12', etc. In that case `Number()` is `NaN`.
+    const versionNumber = Number(window.READTHEDOCS_DATA.version);
+    debug = !Number.isNaN(versionNumber);
+  }
+  return debug;
+}
+/**
+ * Configure Algolia DocSearch.
+ * See https://github.com/algolia/docsearch-configs/blob/master/configs/wagtail.json for index configuration.
+ */
+function docSearchReady() {
+  try {
```

Comment:
The fact we now have to wrap this code in a `try…catch` looks like an anti-pattern to me. Can we deal with the errors that are arising here another way?